Lib/test/test_defaultdict.py

```diff
@@ -183,5 +183,35 @@ def test_pickling(self):
             o = pickle.loads(s)
             self.assertEqual(d, o)
 
+    def test_union(self):
+        i = defaultdict(int, {1: 1, 2: 2})
+        s = defaultdict(str, {0: "zero", 1: "one"})
+
+        self.assertIs((i | s).default_factory, int)
+        self.assertDictEqual(i | s, {1: "one", 2: 2, 0: "zero"})
+        self.assertEqual(list(i | s), [1, 2, 0])
+        self.assertIs((s | i).default_factory, str)
```

Comment:
Maybe introduce an intermediate variable, e.g. `i_s = i | s` to make it abundantly clear that these three assertions are about the same result? Etc.